core/utils/channels.py

```diff
@@ -111,29 +111,24 @@ def async_get_all_current_game_channels():
 
 
 @sync_to_async
-def async_get_game_channel_members(channel: GameChannel) -> List[CustomUser]:
+def async_get_game_channel_members(channel: GameChannel) -> List[GameChannelMember]:
     """Given a game channel object retrieve its expected membership list"""
-    channel.refresh_from_db()
-    queryset = channel.members.all()
-    return list(queryset)  # force evaluation before leaving this sync context
-
-
-def get_baseline_channel_membership(channel: GameChannel) -> List[CustomUser]:
-    game = channel.game
-
-    users = [game.dm.user]
-    party = game.players.filter(standby=False)
-    for player in party:
-        users.append(player.user)
-    return users
-
+    queryset = channel.members.through.objects.filter(channel=channel).prefetch_related("user")
+    result = list(queryset)  # force evaluation before leaving this sync context
+    return result
 
 @sync_to_async
 def async_set_default_channel_membership(channel: GameChannel) -> bool:
     """Attempt to set a default membership list for a game channel"""
+
     try:
-        users = get_baseline_channel_membership(channel)
-        channel.members.set(users)
+        game = channel.game
+        channel.members.set([])
+        channel.members.add(game.dm.user, through_defaults={"manage_messages": True})
```

Comment:
This add channel membership logic is probably going to be reused as we move towards event driven logic, so might be prudent to pull it out to a util function. We can make it a little more readable in the process

Reply:
Unless that function already exists somewhere, which tbh it probably should.